### ISSUE TITLE
The next_target() method of the SSim prototype is miscalculating keepout for observation start times 

### ISSUE THREAD

**Issue by walker-dula** [dsavransky/EXOSIMS#230]
**Describe the bug**
The next_target method only calculates the keep out for the first element of the target list.

Seen Below:
```

        # 2.5 Filter stars not observable at startTimes
        try:
            koTimeInd = np.where(np.round(startTimes[0].value)-self.koTimes.value==0)[0][0]  # find indice where koTime is startTime[0]
            #wherever koMap is 1, the target is observable
            sInds = sInds[np.where(np.transpose(self.koMap)[koTimeInd].astype(bool)[sInds])[0]]# filters inds by koMap #verified against v1.35
        except:#If there are no target stars to observe 
            sInds = np.asarray([],dtype=int)
```

**To Reproduce**
Simply run any SSim case that uses the prototype.

**Expected behavior**
We should be calculating keep out for each individual star.

**EXOSIMS version:**
 - Problem exists in most recent master branch

**Additional context**
The KO calculation appears to be fine for observation endtime:
```
        # 5.2 find spacecraft orbital END positions (for each candidate target), 
        # and filter out unavailable targets
        if len(sInds.tolist()) > 0 and Obs.checkKeepoutEnd:
            try: # endTimes may exist past koTimes so we have an exception to hand this case
                tmpIndsbool = list()
                for i in np.arange(len(sInds)):
                    koTimeInd = np.where(np.round(endTimes[sInds[i]].value)-self.koTimes.value==0)[0][0] # find indice where koTime is endTime[0]
                    tmpIndsbool.append(self.koMap[sInds[i]][koTimeInd].astype(bool)) #Is star observable at time ind
                sInds = sInds[tmpIndsbool]
                del tmpIndsbool
            except:
                sInds = np.asarray([],dtype=int)
```

**Comment by dsavransky** [comment]
@walker-dula please confirm that the latest pull fixes your issue.  @gjsoto check compatibility of this fix with your new multiple keepout maps and adjust as needed.

**Comment by walker-dula** [comment]
This looks identical to my own quick fix that I ran which appears to work. I have not attempted anything with the multiple komaps yet though.

**Comment by dsavransky** [comment]
Fixed in 3.0.0. Closing.
